Makefile

```diff
@@ -1,13 +1,27 @@
 MAKEFLAGS += --silent
 GOLANGCI_LINT_VERSION = $(shell head -n 1 .golangci.yml | tr -d '\# ')
 PROTOC_VERSION := 21.12
-PROTOC_ARCHIVE := protoc-$(PROTOC_VERSION)-linux-x86_64.zip
+
+ifeq ($(OS),Windows_NT)
+    DETECTED_OS := Windows
+    PROTOC_ARCHIVE := protoc-$(PROTOC_VERSION)-win64.zip
```

Comment:
Maybe someone with more Windows expertise can help here, because I don't really know how to set it up correctly. From what I've gathered:

- The `OS` environment variable is always present on Windows, so it's safe to use.
- I tested this using MSYS2 with a native Go installation, not the MSYS2 Go package.

Also, I didn't differentiate between 32-bit and 64-bit Windows since we don't do that for Linux either, but it could be added if needed. But in the 64-bit VM is working fine.